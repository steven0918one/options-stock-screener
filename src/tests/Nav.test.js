import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Nav from '../components/Nav';
import Wrapper from './Wrapper';

Enzyme.configure({ adapter: new Adapter() });

test('Nav should match snapshot', () => {
  const component = renderer.create(
    <Wrapper>
      <Nav />
    </Wrapper>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Nav', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Wrapper>
        <Nav />
      </Wrapper>,
    );
  });
  it('is receiving and running JS', () => {
    expect(2 + 3).toEqual(5);
  });

  it('renders', () => {
    expect(wrapper).not.toBeNull();
  });

  it('starts with an empty input', () => {
    expect(wrapper.find('input').text()).toEqual('');
  });
});
