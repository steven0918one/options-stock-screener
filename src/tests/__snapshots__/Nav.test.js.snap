// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav is rendered 1`] = `
<nav>
  <ul>
    <div
      className="logo"
    />
    <a
      href="/"
      onClick={[Function]}
    >
      <li
        id="home"
      >
        Market Overview
      </li>
    </a>
    <a
      href="/details"
      onClick={[Function]}
    >
      <li
        id="details"
      >
        Stock Details
      </li>
    </a>
    <li>
      <div
        className="form-container-nav"
      >
        <form
          className="books-form"
          onSubmit={[Function]}
        >
          <input
            className="title-input"
            name="title"
            onChange={[Function]}
            placeholder="Search Symbol"
            type="text"
          />
          <button
            type="submit"
          >
            <i
              className="fas fa-chart-line"
            />
          </button>
        </form>
      </div>
    </li>
  </ul>
</nav>
`;

exports[`Nav should match snapshot 1`] = `
<nav>
  <ul>
    <div
      className="logo"
    />
    <a
      href="/"
      onClick={[Function]}
    >
      <li
        id="home"
      >
        Market Overview
      </li>
    </a>
    <a
      href="/details"
      onClick={[Function]}
    >
      <li
        id="details"
      >
        Stock Details
      </li>
    </a>
    <li>
      <div
        className="form-container-nav"
      >
        <form
          className="books-form"
          onSubmit={[Function]}
        >
          <input
            className="title-input"
            name="title"
            onChange={[Function]}
            placeholder="Search Symbol"
            type="text"
          />
          <button
            type="submit"
          >
            <i
              className="fas fa-chart-line"
            />
          </button>
        </form>
      </div>
    </li>
  </ul>
</nav>
`;
